cluster-autoscaler/core/scaledown/planner/planner.go

```diff
@@ -289,15 +296,34 @@ func (p *Planner) categorizeNodes(podDestinations map[string]bool, scaleDownCand
 		// TODO(x13n): break on timeout. Figure out how to handle nodes
 		// identified as unneeded in previous iteration, but now
 		// skipped due to timeout.
-		removable, unremovable := p.rs.SimulateNodeRemoval(node, podDestinations, p.latestUpdate, pdbs)
-		if unremovable != nil {
-			unremovableCount += 1
-			p.unremovableNodes.AddTimeout(unremovable, unremovableTimeout)
-		}
+		removable, unremovable := p.rs.SimulateNodeRemoval(node, podDestinations, p.latestUpdate, p.pdbRemainingDisruptions.GetPdbs())
 		if removable != nil {
+			ok, isRisky, blockingPod := p.pdbRemainingDisruptions.CanDisrupt(removable.PodsToReschedule)
+			if !ok {
+				if blockingPod != nil {
+					unremovable = &simulator.UnremovableNode{
```

Comment:
There is quite a lot of nesting here, can you move some of it to a helper function?

Reply:
I removed `if blockingPod != nil` block, since now for removable pod we need to check only isRisky value, seems it looks better now, WDYT?

Reply:
I'm a fan of short functions with meaningful names, but won't insist, this looks good enough now.